### ISSUE TITLE
💅 `nursery/noJsxLiterals` does not respect whitespace around `allowedStrings`

### ISSUE THREAD

**Issue by LeCarbonator** [biomejs/biome#7795]
### Environment information

```bash
CLI:
  Version:                      2.2.6
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.15.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/11.4.1

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  false
  Enabled rules:
    nursery/noJsxLiterals

Workspace:
  Open Documents:               0
```

### Rule name

nursery/noJsxLiterals

### Playground link

https://github.com/LeCarbonator/biome-repro-noJsxLiterals

### Expected result

These are expected to both not error because only allowed strings are used as jsx literals.
```bash
npm run biome
npm run eslint
```
However, biome will show an error for one of the snippets.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
